File: src/main/cpp/subsystems/CameraSubsystem.cpp
#include <subsystems/CameraSubsystem.h>
#include <frc/smartdashboard/SmartDashboard.h>

CameraSubsystem::CameraSubsystem(){
    inst = nt::NetworkTableInstance::GetDefault();
    table = inst.GetTable("aprilTags");
    centerX = table->GetDoubleTopic("centerX").Subscribe(0.0);
    centerY = table->GetDoubleTopic("centerY").Subscribe(0.0);
    tagId = table->GetIntegerTopic("tagId").Subscribe(0.0);
    detection = table->GetBooleanTopic("detection").Subscribe(false);
    height = table->GetDoubleTopic("height").Subscribe(0.0);
}

void CameraSubsystem::Periodic(){
    
}

void CameraSubsystem::PutStuffOnSmartDashboard(){
    frc::SmartDashboard::PutNumber("centerX", centerX.Get());
    frc::SmartDashboard::PutNumber("centerY", centerY.Get());
    frc::SmartDashboard::PutNumber("tagId", tagId.Get());
    frc::SmartDashboard::PutBoolean("detection", detection.Get());
    frc::SmartDashboard::PutNumber("height", height.Get());
}

File: src/main/java/frc/robot/subsystems/CameraSubsystem.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.subsystems;

import edu.wpi.first.networktables.BooleanSubscriber;
import edu.wpi.first.networktables.DoubleSubscriber;
import edu.wpi.first.networktables.IntegerSubscriber;
import edu.wpi.first.networktables.NetworkTable;
import edu.wpi.first.networktables.NetworkTableInstance;
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
import edu.wpi.first.wpilibj2.command.SubsystemBase;

public class CameraSubsystem extends SubsystemBase {
    private final NetworkTableInstance inst;
    private final NetworkTable table;

    public final BooleanSubscriber detection;
    public final IntegerSubscriber tagId;
    public final DoubleSubscriber distance;
    public final DoubleSubscriber yaw;

    /** Creates a new CameraSubsystem. */
    public CameraSubsystem() {
        inst = NetworkTableInstance.getDefault();
        table = inst.getTable("aprilTags");
        tagId = table.getIntegerTopic("tagId").subscribe(0);
        detection = table.getBooleanTopic("detection").subscribe(false);
        distance = table.getDoubleTopic("distance").subscribe(0.0);
        yaw = table.getDoubleTopic("yaw").subscribe(0.0);
    }

    @Override
    public void periodic() {
        // This method will be called once per scheduler run
    }

    public void putStuffOnSmartDashboard() {
        SmartDashboard.putNumber("tagId", tagId.get());
        SmartDashboard.putBoolean("detection", detection.get());
        SmartDashboard.putNumber("distance", distance.get());
        SmartDashboard.putNumber("yaw", yaw.get());
    }

    public double getYaw() {
        return yaw.get();
    }
}
